src/main/scala/za/co/absa/spark/commons/test/YarnSparkConfiguration.scala

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2021 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.spark.commons.test
+
+import java.io.File
+import java.time.ZoneId
+import java.util.TimeZone
+
+import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.fs.{FileSystem, Path}
+import org.apache.spark.SparkConf
+import org.apache.spark.sql.SparkSession
+
+import scala.collection.JavaConversions.iterableAsScalaIterable
+
+class YarnSparkConfiguration(confDir: String, distJarsDir: String) extends SparkTestConfig {
+
+  override protected def master: String = "yarn"
+
+  override def appName: String = "Yarn Testing"
+
+  override def sparkSession = SparkSession.builder().config(new SparkConf().setAll(getHadoopConfigurationForSpark(confDir)))
+    .config("spark.yarn.jars", getDependencies())
+    .config("spark.deploy.mode", "client")
+    .getOrCreate()
```

Comment:
Levereging the class structure:
```suggestion
  override protected def builder: SparkSession.Builder = {
    super.builder
    .setAll(getHadoopConfigurationForSpark(confDir)))
    .config(new SparkConf().setAll(getHadoopConfigurationForSpark(confDir)))
    .config("spark.yarn.jars", getDependencies())
    .config("spark.deploy.mode", "client")
    .getOrCreate()
```